### DISCUSSION TITLE
Displacment values are not showing correctly as expected

### DISCUSSION THREAD

**Discussion by Narayanan1999** [idaholab/moose#27265] [Q&A General]
Hi,
I tried to do  displacement analysis by pulling a plate from two opposites side on X axis by 0.001. Using Bond Based Peridynamics and horizon number =1, the displacment along X direction is coming correctly as expected. But displacement along Y direction is obtained as **zero** irrespective of what the material property is. It seems like Problems are treated as truss structures as element type is EDGE2. What can I do?
The Code written is as follows. 
```
elem = QUAD4
[GlobalParams]
  displacements = 'disp_x disp_y'
[]

L=1.0
B=1.0
Hor_num=1
dimension=2
xn=10
yn=10
delx=${fparse L/xn}
dely=${fparse L/xn}
M_L_min=${fparse -delx/2}
M_L_max=${fparse L+delx/2}
M_B_min=${fparse -dely/2}
M_B_max=${fparse B+dely/2}
M_nx=${fparse xn+1}
M_ny=${fparse yn+1}
YM=1000
PR=0.3
[Mesh]
  type = PeridynamicsMesh
  horizon_number = ${Hor_num}
  [Mainsec]
    type = GeneratedMeshGenerator
    dim = ${dimension}
    xmin = ${M_L_min}
    xmax = ${M_L_max}
    ymin = ${M_B_min}
    ymax = ${M_B_max}
    nx = ${M_nx}
    ny = ${M_ny}
    elem_type = ${elem}
    boundary_name_prefix = Mainsec
    boundary_id_offset = 10
  []
  [./gpd]
    type = MeshGeneratorPD
    input = Mainsec
    retain_fe_mesh = false
  [../]
[]

[Variables]
  [./disp_x]
  [../]
  [./disp_y]
  [../]
[]

[BCs]
  [./left_x1]
    type = FunctionDirichletBC
    variable = disp_x
    boundary = 1013
    function = '-0.001 * t'
  [../]
  [./right_x1]
    type = FunctionDirichletBC
    variable = disp_x
    boundary = 1011
    function = '0.001 * t'
  [../]
[]

[Modules/Peridynamics/Mechanics/Master]
  [./all]
    formulation = BOND
  [../]
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = ${YM}
    poissons_ratio = ${PR}
  [../]

  [./force_density]
    type =  ComputeSmallStrainConstantHorizonMaterialBPD
	plane_stress=true
  [../]
[]

[Preconditioning]
  [./SMP]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  end_time = 1
  dt = 1
  num_steps = 1
[]
[Outputs]
  exodus = true
[]
 ```

**Comment by GiudGiud** [comment]
@hchen139 @laagesen

**Reply by laagesen** [reply]
Sorry, afraid peridynamics is not in my wheelhouse. I think a lot of this capability was added by someone who is no longer at INL, I cannot remember the name but @bwspenc might remember who it was

**Comment by hchen139** [comment]
Briefly looked at your input file. There are at least two major issues. First, using _Hor_num=1_ indicates you only consider the interaction with the nearest four neighboring elements. A recommended value is 3. Second, you didn't impose sufficient boundary conditions to fully constrain your problem. What is/are the constraint(s) for disp_y?

**Reply by Narayanan1999** [reply]
For __Hor_num_=3_, I tried changing Poisson's ratio and also tried plane stress case, but displacement values along Y-direction obtained is found to be same irrespective of these changes in material properties.
